packages/server/graphql/mutations/helpers/addDiscussionTopics.ts

```diff
@@ -11,8 +11,22 @@ const addDiscussionTopics = async (meeting: RetrospectiveMeeting, dataLoader: Da
   if (!discussPhase) return {discussPhaseStages: [], meetingId}
   const placeholderStage = discussPhase.stages[0]
   if (!placeholderStage) return {discussPhaseStages: [], meetingId}
-  const reflectionGroups = await dataLoader.get('retroReflectionGroupsByMeetingId').load(meetingId)
-  const sortedReflectionGroups = mapGroupsToStages(reflectionGroups)
+
+  const [reflectionGroups, reflections] = await Promise.all([
+    dataLoader.get('retroReflectionGroupsByMeetingId').load(meetingId),
+    dataLoader.get('retroReflectionsByMeetingId').load(meetingId)
+  ])
+
+  const groupReflectionCounts = reflections.reduce((counts: Record<string, number>, reflection) => {
+    counts[reflection.reflectionGroupId] = (counts[reflection.reflectionGroupId] || 0) + 1
+    return counts
+  }, {})
+
+  const activeGroups = reflectionGroups.filter(
+    (group) => (groupReflectionCounts[group.id] ?? 0) > 0
+  )
```

Comment:
-1 we already call `removeEmptyReflections` in `handleCompletedRetrospectiveStage`, https://github.com/ParabolInc/parabol/blob/09aa43056087b089310dd9a0cacf1103b9c67609/packages/server/graphql/mutations/helpers/handleCompletedStage.ts#L28
We should also remove empty reflection groups there, i.e. set them to `isActive: false`.
Otherwise if we cannot rely on this field being accurate when calling the `retroReflectionGroupsByMeetingId` loader, then we should get rid of the field completely and modify the loader to do a join with reflections to check there are some in the group.

Reply:
Thanks, makes sense. I've updated `removeEmptyReflections` to set reflection groups to be inactive if they don't contact any reflections

Reply:
Thanks, removed!